notebooks/python/neo4j/README.md

```diff
@@ -0,0 +1,253 @@
+
+    floatCastSelectExpr=["NAME as name","cast(POPULATION as long) population",\
+    "cast(PERCENTOVER25WITHBACHELORS as float) percentOver25WithBachelors",\
+    "cast(MEDIANHOMEPRICE as int) medianHomePrice",\
+    "cast(MEDIANHOUSEHOLDINCOME as int) medianHouseholdIncome"]
+
+    castedDf = rawDf.selectExpr(floatCastSelectExpr)
+
+After down-casting doubles and decimals which Neo4j does not support, to floats or integers which it does, populating Neo4j nodes from a Spark DataFrame is trivial, and identical regardless of source.
+
+    (castedDf.write
+    .format("org.neo4j.spark.DataSource")
+    .mode("ErrorIfExists")
+    .option("labels", ":MSA")
+    .option("url", neo4j_url)
+    .option("authentication.type", "basic")
+    .option("authentication.basic.username", neo4j_user)
+    .option("authentication.basic.password", neo4j_password)
+    .option("partitions", "1")
+    .save())
+
+BigQuery fully supports Neo4j datatypes but casting may be necessary if the native types are mismatched.  
+
+## Big Query
+Also, BigQuery users know that plain SQL queries require `viewsEnabled` and `materializationDataset` options to be set.  
+
+Every time a query is executed, intermediary results will be cached in a temporary table in the schema (BigQuery calls it a “dataset”).
+
+That table will be reaped after 1 day by default, more frequently if overridden.
+
+    spark.conf.set("viewsEnabled","true")
+    spark.conf.set("materializationDataset","census")
+    # if this value is too low, big query will expect that it exists in cache.  Default is 1440.
+    spark.conf.set("materializationExpirationTimeInMinutes",5)
+    
+    msa_query="""
+        SELECT name
+        ,population
+        ,percentOver25WithBachelors
+        ,medianHomePrice
+        ,medianHouseholdIncome
+        FROM `neo4jbusinessdev.census.msa_demos`
+    """
+    bigqueryDf = spark.read.format("bigquery").option("query",msa_query).load()
+
+However, materialization is not necessary when using the table (vs query) option because a BigQuery likes to keep around result caches but a table (SELECT * FROM <table>) doesn’t need a copy of itself!  For example:
+
+    msa_table="""
+        neo4jbusinessdev.census.msa_demos
+    """
+    tableBqDf = spark.read.format("bigquery").option("table",msa_table).load()
+
+Should derivative DataFrames then be required, secondary filtering can be done using Spark SQL as shown:
+
+    tableBqDf.createOrReplaceTempView("msa_df")
+    bigqueryPostQueryDf = spark.sql("SELECT * FROM msa_df WHERE population > 1000")
+
+After enriching features in a graph, it may be desirable to write them back to a relational source for use downstream in an AI/ML pipeline, or with business intelligence clients like Tableau, Looker, or PowerBI.
+
+With BigQuery, a temporary Google Cloud Storage (GCS) bucket is required.  Otherwise, everything is straight-forward in the Spark DataFrame way.
+
+    # tmp_working_bucket
```

Comment:
Please surround with code block and fix indentation

Also - why is the statement surrounded with parenthesis?